lib/Doctrine/ODM/MongoDB/DocumentManager.php

```diff
@@ -842,4 +858,18 @@ public function getFilterCollection()
 
         return $this->filterCollection;
     }
+
+    private function arrayContainsDocuments(array $documentOrOptions)
+    {
+        foreach ($documentOrOptions as $documentPerhaps) {
+            if (! is_object($documentPerhaps)) {
+                return false;
```

Comment:
Any reason why you decided to treat the array as options upon encountering a non-object value? Given what we expect, I'd actually say that unless none of the given values is a transient document, we may want to treat it as such and keep the original behaviour.

```suggestion
                continue;
```

Reply:
Reasoning is that any and all documents must be an object in the first place. So if it's not an object, something would have failed down the line anyway as unit of work would be trying to compute change set or an array or scalar

Reply:
Ok, that makes sense. We can keep it this way then